src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java

```diff
@@ -117,6 +125,31 @@ public MetricsLogger setDimensions(DimensionSet... dimensionSets) {
         return this;
     }
 
+    /**
+     * Overwrite custom dimensions on this MetricsLogger instance, with an option to preserve
+     * default dimensions.
+     *
+     * @param preserveDefault indicates whether default dimensions should be used
+     * @param dimensionSets the dimensionSets to set
+     * @return the current logger
+     */
+    public MetricsLogger setDimensions(boolean preserveDefault, DimensionSet... dimensionSets) {
```

Comment:
Would it make sense to change the order of parameters from `setDimensions(preserveDefault,  dimensionSets)` to `setDimensions(dimensionSets, preserveDefault)`?